include/drivers/usbc/usbc_tcpc.h

```diff
@@ -0,0 +1,1067 @@
+	TCPC_ALERT_POWER_STATUS,
+	/** Receive Buffer register changed */
+	TCPC_ALERT_MSG_STATUS,
+	/** Received Hard Reset message */
+	TCPC_ALERT_HARD_RESET_RECEIVED,
+	/** SOP* message transmission not successful */
+	TCPC_ALERT_TRANSMIT_MSG_FAILED,
+	/**
+	 * Reset or SOP* message transmission not sent
+	 * due to an incoming receive message
+	 */
+	TCPC_ALERT_TRANSMIT_MSG_DISCARDED,
+	/** Reset or SOP* message transmission successful */
+	TCPC_ALERT_TRANSMIT_MSG_SUCCESS,
+	/** A high-voltage alarm has occurred */
+	TCPC_ALERT_VBUS_ALARM_HI,
+	/** A low-voltage alarm has occurred */
+	TCPC_ALERT_VBUS_ALARM_LO,
+	/** A fault has occurred. Read the FAULT_STATUS register */
+	TCPC_ALERT_FAULT_STATUS,
+	/** TCPC RX buffer has overflowed */
+	TCPC_ALERT_RX_BUFFER_OVERFLOW,
+	/** The TCPC in Attached.SNK state has detected a sink disconnect */
+	TCPC_ALERT_VBUS_SNK_DISCONNECT,
+	/** Receive buffer register changed */
+	TCPC_ALERT_BEGINNING_MSG_STATUS,
+	/** Extended status changed */
+	TCPC_ALERT_EXTENDED_STATUS,
+	/**
+	 * An extended interrupt event has occurred. Read the alert_extended
+	 * register
+	 */
+	TCPC_ALERT_EXTENDED,
+	/** A vendor defined alert has been detected */
+	TCPC_ALERT_VENDOR_DEFINED
+};
+
+/**
+ * @brief TCPC Status register
+ */
+enum tcpc_status_reg {
+	/** The CC Status register */
+	TCPC_CC_STATUS,
+	/** The Power Status register */
+	TCPC_POWER_STATUS,
+	/** The Fault Status register */
+	TCPC_FAULT_STATUS,
+	/** The Extended Status register */
+	TCPC_EXTENDED_STATUS,
+	/** The Extended Alert Status register */
+	TCPC_EXTENDED_ALERT_STATUS,
+	/** The Vendor Defined Status register */
+	TCPC_VENDOR_DEFINED_STATUS
+};
+
+/**
+ * @brief TCPC Chip Information
+ */
+struct tcpc_chip_info {
+	/** Vendor Id */
+	uint16_t vendor_id;
+	/** Product Id */
+	uint16_t product_id;
+	/** Device Id */
+	uint16_t device_id;
+	/** Firmware version number */
+	uint64_t fw_verion_number;
+
+	union {
+		/** Minimum Required firmware version string */
+		uint8_t min_req_fw_version_string[8];
+		/** Minimum Required firmware version number */
+		uint64_t min_req_fw_version_number;
+	};
+};
+
+typedef int (*tcpc_vbus_cb_t)(const struct device *dev, int *vbus_meas);
+typedef int (*tcpc_discharge_vbus_cb_t)(const struct device *dev,
+					bool enable);
+typedef int (*tcpc_vconn_control_cb_t)(const struct device *dev,
+					bool enable);
+typedef void (*tcpc_alert_handler_cb_t)(const struct device *dev,
+					void *data,
+					enum tcpc_alert alert);
+
+typedef int (*tcpc_init_t)(const struct device *dev);
+typedef int (*tcpc_get_cc_t)(const struct device *dev,
+			     enum tc_cc_voltage_state *cc1,
+			     enum tc_cc_voltage_state *cc2);
+typedef void (*tcpc_set_vbus_measure_cb_t)(const struct device *dev,
+					   tcpc_vbus_cb_t vbus_cb);
+typedef void (*tcpc_set_discharge_vbus_cb_t)(const struct device *dev,
+					     tcpc_discharge_vbus_cb_t discharge_vbus_cb);
+typedef bool (*tcpc_check_vbus_level_t)(const struct device *dev,
```

Comment:
I guess all of these not `*_cb_t `typedefs are not necessary.

Reply:
I thought using typedefs would make it easier to read the struct tcpc_driver_api {

Reply:
I disagree. Besides, these are only used in this header fine, so they are not necessary.